core/src/modules/ops/destination.ts

```diff
@@ -912,82 +923,109 @@ export namespace DestinationOps {
     success: boolean;
     error: Error;
   }> {
-    const _exports: Export[] = []; // only ones we are sending
-    for (const _export of givenExports) {
-      // TODO: maybe should recalcuate hasChanges and from current mostRecent
-      if (!_export.hasChanges) {
-        await _export.complete(); // do not include exports with hasChanges=false
-      } else {
-        _exports.push(_export);
-      }
-    }
+    const _exports: Export[] = [];
+    const locks: Awaited<ReturnType<typeof getLock>>[] = [];
 
-    const exportRecords: ExportRecordsPluginMethod = await getBatchFunction(
-      destination
-    );
+    try {
+      for (const givenExport of givenExports) {
+        if (!givenExport.hasChanges) await givenExport.complete();
 
-    const syncMode = await destination.getSyncMode();
-    const syncOperations: DestinationSyncOperations = syncMode
-      ? DestinationSyncModeData[syncMode].operations
-      : DestinationSyncModeData.sync.operations; // if destination does not support sync modes, allow all
+        const lock = await getLock(
+          `${givenExport.recordId}:${givenExport.destinationId}`
+        );
 
-    const options = await destination.getOptions();
-    const app = await destination.$get("app", {
-      scope: null,
-      include: [Option],
-    });
-    const appOptions = await app.getOptions();
-    const connection = await app.getConnection();
+        const mostRecentExport = await Export.findOne({
+          where: {
+            recordId: givenExport.recordId,
+            destinationId: destination.id,
+          },
+          order: [["createdAt", "DESC"]],
+        });
+
+        const isNewest = mostRecentExport.id === givenExport.id;
+        if (!isNewest) console.info(mostRecentExport);
+        const gotLock = typeof lock === "function";
+
+        if (gotLock) locks.push(lock);
+        if (!isNewest) {
+          await cancelOldExport(givenExport, mostRecentExport);
```

Comment:
I like canceling either way. Maybe do this before getting the lock if we are going to do it either way and bail. One less thing to unlock.

Reply:
👍 agree - might as well always cancel old exports.